src/Illuminate/View/Compilers/Concerns/CompilesJson.php

```diff
@@ -19,12 +19,90 @@ trait CompilesJson
      */
     protected function compileJson($expression)
     {
-        $parts = explode(',', $this->stripParentheses($expression));
+        $parts = $this->parseArguments($this->stripParentheses($expression));
 
         $options = isset($parts[1]) ? trim($parts[1]) : $this->encodingOptions;
 
         $depth = isset($parts[2]) ? trim($parts[2]) : 512;
 
-        return "<?php echo json_encode($parts[0], $options, $depth) ?>";
+        // Ensure we have at least one argument, default to null if empty
+        $data = $parts[0] ?? 'null';
+
+        return "<?php echo json_encode($data, $options, $depth) ?>";
+    }
+
+    /**
+     * Parse arguments from an expression, respecting nested structures.
+     *
+     * This method properly handles commas inside arrays, closures, function calls,
+     * and other nested structures by using PHP's tokenizer.
+     *
+     * @param  string  $expression
+     * @return array
```

Comment:
You can narrow this further down:

```suggestion
     * @return array{0?: string, 1?: string, 2?: string}
```

Reply:
you're welcome!